vmdk_plugin/vmdkops/esx_vmdkcmd.go

```diff
@@ -85,19 +86,30 @@ func (vmdkCmd EsxVmdkCmd) Run(cmd string, name string, opts map[string]string) (
 	defer C.free(unsafe.Pointer(ans))
 
 	// connect, send command, get reply, disconnect - all in one shot
-	_, err = C.Vmci_GetReply(C.int(EsxPort), cmdS, beS, ans)
-
-	if err != nil {
-		var errno syscall.Errno
-		errno = err.(syscall.Errno)
-		msg := fmt.Sprintf("'%s' failed: %v (errno=%d).", cmd, err, int(errno))
-		if errno == syscall.ECONNRESET {
-			msg += " Hit communication issue with ESX (vmci or ESX service)\n"
-			msg += " Please refer to the FAQ https://github.com/vmware/docker-volume-vsphere/wiki#faq"
+	retryCount := 10
```

Comment:
Oops, absolutely right. Sorry, its fine as is.

On Thu, Jun 30, 2016 at 1:46 PM, Mark Sterin <notifications@github.com>
wrote:

> In vmdk_plugin/vmdkops/esx_vmdkcmd.go
> <https://github.com/vmware/docker-volume-vsphere/pull/499#discussion_r69090189>
> :
>
> > @@ -85,19 +86,30 @@ func (vmdkCmd EsxVmdkCmd) Run(cmd string, name string, opts map[string]string) (
> >  	defer C.free(unsafe.Pointer(ans))
> >
> >  	// connect, send command, get reply, disconnect - all in one shot
> > -	_, err = C.Vmci_GetReply(C.int(EsxPort), cmdS, beS, ans)
> > -
> > -	if err != nil {
> > -		var errno syscall.Errno
> > -		errno = err.(syscall.Errno)
> > -		msg := fmt.Sprintf("'%s' failed: %v (errno=%d).", cmd, err, int(errno))
> > -		if errno == syscall.ECONNRESET {
> > -			msg += " Hit communication issue with ESX (vmci or ESX service)\n"
> > -			msg += " Please refer to the FAQ https://github.com/vmware/docker-volume-vsphere/wiki#faq"
> > +	retryCount := 10
>
> @govint <https://github.com/govint> - please do check the code before
> posting comments. The disk_attach() code says:
>
>   # Check if this disk is already attached, and if it is - skip the attach
>     device = findDeviceByPath(vmdk_path, vm)
>     if device:
>         # Disk is already attached.
>         logging.warning("Disk %s already attached. VM=%s",
>                         vmdk_path, vm.config.uuid)
>         setStatusAttached(vmdk_path, vm)
>         return busInfo(device.unitNumber,
>                        device.controllerKey - offset_from_bus_number)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/499/files/23e50189635c854f88aaf3f885bcc0188045c2ed#r69090189>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseF5iKX0gTX2CxDCifrqQOi_Xp5_Pks5qQ3tdgaJpZM4JBwl3>
> .
>